### PR TITLE
Add notify if fails slack monitoring of barman backup process

### PR DESCRIPTION
**Pull request by mikethebeer**

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -27,17 +27,19 @@ RUN cd /usr/lib/python3.9 && tar -zxvf /python-libs.tgz
...
+    cron jq curl barman barman-cli barman-cli-cloud && \
     apt-get clean
 
 COPY psqlrc /root/.psqlrc
 
 ENV PGHOST=${PGHOST:-127.0.0.1} \
     PGUSER=${PGUSER:-postgres} \
-    BARMAN_BACKUP_SCHEDULE="0 4 * * *"
+    BARMAN_BACKUP_SCHEDULE="0 4 * * *" \
+    NOTIFY_BACKUP_ERROR_MESSAGE="Backup failed"
```
Comment (mikethebeer) [inline docker/Dockerfile:42]: using $PGHOST or $PGUSER here results in empty string. Not sure how to evauate these environment vars during build.

Comment (frisi) [reply docker/Dockerfile:42]: at least locally in bash this syntax should work
```
frisi@frisi-xps:~$ export PGHOST=localhost
frisi@frisi-xps:~$ echo "hallo ${MESSAGE:-backup failed for $PGHOST}"
hallo backup failed for localhost
frisi@frisi-xps:~$ export MESSAGE="custom message"
frisi@frisi-xps:~$ echo "hallo ${MESSAGE:-backup failed for $PGHOST}"
hallo custom message
```

File: docker/entrypoint.sh [hunk 2]
```diff
@@ -2,7 +2,7 @@
 set -e
 
 echo "Generating cron schedules"
-echo "${BARMAN_BACKUP_SCHEDULE} barman /usr/local/bin/barman backup --wait all" >> /etc/cron.d/barman
+echo "$BARMAN_BACKUP_SCHEDULE barman /usr/bin/notify_if_fails '/usr/bin/barman backup --wait all' '$NOTIFY_BACKUP_ERROR_MESSAGE'" >> /etc/cron.d/barman
```
Comment (frisi) [inline docker/entrypoint.sh:5]:
```suggestion
echo "$BARMAN_BACKUP_SCHEDULE barman /usr/bin/notify_if_fails '/usr/bin/barman backup --wait all' '${NOTIFY_BACKUP_ERROR_MESSAGE:-barman backup failed for $PGUSER@$PGHOST}'" >> /etc/cron.d/barman
```

**Review by frisi — APPROVED** [review]
LGTM.
to be honest, i don't understand why when running the crontab `$NOTIFY_BACKUP_ERROR_MESSAGE` should be available whilst `$PGHOST` is not.

**Review by frisi — CHANGES_REQUESTED** [review]
evtl add NOTIFY_BACKUP_ERROR_MESSAGE and also the slack_cli_token to docker-compose as minimal documentation. 
esp slack_cli_token is currently totally undocumented

File: examples/backup/docker-compose.yml [hunk 3]
```diff
@@ -54,7 +54,9 @@ services:
       PGHOST: postgres
       AWS_ACCESS_KEY_ID: minioadmin
       AWS_SECRET_ACCESS_KEY: minioadmin
-
+      BARMAN_BACKUP_SCHEDULE: 0 4 * * *
+      NOTIFY_BACKUP_ERROR_MESSAGE: "Test: Backup failed"
```
Comment (frisi) [inline examples/backup/docker-compose.yml:58]:
```suggestion
      NOTIFY_BACKUP_ERROR_MESSAGE: "Testproject: Backup failed"
```

**Review by frisi — APPROVED** [review]

**Pull request merged**